File: services/delivery/src/test/groovy/io/wkrzywiec/fooddelivery/delivery/domain/ItemTestData.java
package io.wkrzywiec.fooddelivery.delivery.domain;

import lombok.Getter;

import java.math.BigDecimal;

@Getter
public class ItemTestData {

    private String name = "Pizza Margherita";
    private int amount = 1;
    private BigDecimal pricePerItem = new BigDecimal(10);

    private ItemTestData() {};

    Item entity() {
        return Item.builder()
                .name(name)
                .amount(amount)
                .pricePerItem(pricePerItem)
                .build();
    }

    public io.wkrzywiec.fooddelivery.delivery.domain.incoming.Item dto() {
        return new io.wkrzywiec.fooddelivery.delivery.domain.incoming.Item(name, amount, pricePerItem);
    }

    public static ItemTestData anItem() {
        return new ItemTestData();
    }

    public ItemTestData withName(String name) {
        this.name = name;
        return this;
    }

    public ItemTestData withPricePerItem(double price) {
        this.pricePerItem = new BigDecimal(price);
        return this;
    }

    ItemTestData withAmount(int amount) {
        this.amount = amount;
        return this;
    }
}

File: services/ordering/src/main/kotlin/io/wkrzywiec/fooddelivery/ordering/domain/ItemTestData.kt
package io.wkrzywiec.fooddelivery.ordering.domain

import java.math.BigDecimal

data class ItemTestData private constructor(
    val name: String = "Pizza Margherita",
    val amount: Int = 1,
    val pricePerItem: BigDecimal = BigDecimal(10)
) {

    companion object {
        @JvmStatic
        fun anItem(): ItemTestData {
            return ItemTestData()
        }
    }

    fun entity(): Item {
        return Item(
            name = name,
            amount = amount,
            pricePerItem = pricePerItem
        )
    }

    fun dto(): io.wkrzywiec.fooddelivery.commons.model.Item {
        return io.wkrzywiec.fooddelivery.commons.model.Item(name, amount, pricePerItem)
    }

    fun withName(name: String): ItemTestData {
        return this.copy(name = name)
    }

    fun withPricePerItem(price: Double): ItemTestData {
        return this.copy(pricePerItem = BigDecimal(price))
    }

    fun withAmount(amount: Int): ItemTestData {
        return this.copy(amount = amount)
    }
}
